core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/model/MemStatement.java

```diff
@@ -66,31 +78,23 @@ public MemStatement(MemResource subject, MemIRI predicate, MemValue object, MemR
 			int sinceSnapshot) {
 		super(subject, predicate, object, context);
 		this.explicit = explicit;
-		setSinceSnapshot(sinceSnapshot);
-	}
-
-	/*---------*
-	 * Methods *
-	 *---------*/
-
-	public void setSinceSnapshot(int snapshot) {
-		sinceSnapshot = snapshot;
+		this.sinceSnapshot = sinceSnapshot;
 	}
 
 	public int getSinceSnapshot() {
 		return sinceSnapshot;
 	}
 
 	public void setTillSnapshot(int snapshot) {
-		tillSnapshot = snapshot;
+		TILL_SNAPSHOT.setRelease(this, snapshot);
 	}
 
 	public int getTillSnapshot() {
-		return tillSnapshot;
+		return (int) TILL_SNAPSHOT.getAcquire(this);
 	}
 
 	public boolean isInSnapshot(int snapshot) {
-		return snapshot >= sinceSnapshot && snapshot < tillSnapshot;
+		return snapshot >= sinceSnapshot && snapshot < ((int) TILL_SNAPSHOT.getAcquire(this));
 	}
```

Comment:
`tilSnapshot` is now updated with release/acquire memory semantics which are faster than pure volatile operations on many CPU architectures.